sre-recipes/recipe_runner.py

```diff
@@ -0,0 +1,196 @@
+        if not self.recipe:
+            raise NotImplementedError(
+                f"No valid implementation exists for `{recipe_name}` recipe.")
+
+    def get_name(self):
+        return self.recipe.get_name()
+
+    def get_description(self):
+        return self.recipe.get_description()
+
+    def run_break(self):
+        return self.recipe.run_break()
+
+    def run_restore(self):
+        return self.recipe.run_restore()
+
+    def run_hint(self):
+        return self.recipe.run_hint()
+
+    def run_verify(self):
+        return self.recipe.run_verify()
+
+
+class ConfigBasedRecipeRunner:
+    """A SRE Recipe runner for running recipes implemented using configs.
+
+    Given a `recipe_name`, it tries to load `recipes/configs_based/recipe_name.yaml`.
+
+    This runner will propgate all exceptions to the caller, and it is caller's
+    responsibility to handle any exception and to perform any error logging.
+    """
+
+    def __init__(self, recipe_name):
+        filepath = path.join(path.dirname(
+            path.abspath(__file__)), f"recipes/configs_based/{recipe_name}.yaml")
+        with open(filepath, "r") as file:
+            self.recipe = yaml.safe_load(file.read())
+        if not self.recipe:
+            raise ValueError("Cannot parse config as YAML.")
+
+    def get_name(self):
+        return self.recipe.get("name", "No name found")
+
+    def get_description(self):
+        return self.recipe.get("description", "No description found")
+
+    ############################ Run Recipe ###################################
+
+    @ property
+    def config(self):
+        return self.recipe.get("config", {})
+
+    def run_break(self):
+        print('Deploying broken service...')
+        self.__handle_actions(self.config.get("break", {}))
+        print('Done. Deployed broken service')
+
+    def run_restore(self):
+        print('Restoring broken service...')
+        self.__handle_actions(self.config.get("restore", {}))
+        print('Done. Restored broken service to working state.')
+
+    def run_hint(self):
+        hint = self.config.get("hint", None)
+        if hint:
+            print(f'Here is your hint!\n\n{hint}')
+        else:
+            print("This recipe has no hints.")
+
+    def run_verify(self):
+        verify_config = self.config.get("verify", {})
+        if not verify_config:
+            raise NotImplementedError("Verify is not configured")
+
+        affected_service_config = verify_config.get("affected_service", {})
+        if affected_service_config:
+            if "answer" not in affected_service_config:
+                raise ValueError(
+                    "Correct answer is not specified for affected service quiz.")
+            elif "choices" not in affected_service_config:
+                raise ValueError(
+                    "No answer choices configured in affected service quiz.")
+            utils.run_interactive_multiple_choice(
+                "Which service has an issue?",
```

Comment:
This may not be important for this PR, but I would like it if we could generalize these questions a bit more. Maybe include the prompt text in the YAML, so we're not always locked in to these same two questions?

Reply:
Done. Same as above. Made it into a new action type.